packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeProofFormatService.ts

```diff
@@ -0,0 +1,309 @@
+
+    ps?.validatePresentationDefinition(presentationDefinition)
+
+    const format = new ProofFormatSpec({ format: PRESENTATION_EXCHANGE_PRESENTATION_PROPOSAL, attachmentId })
+
+    const attachment = this.getFormatData(presentationDefinition, format.attachmentId)
+
+    return { format, attachment }
+  }
+
+  public async processProposal(agentContext: AgentContext, { attachment }: ProofFormatProcessOptions): Promise<void> {
+    const ps = this.presentationExchangeService(agentContext)
+    const proposal = attachment.getDataAsJson<PresentationDefinition>()
+    ps.validatePresentationDefinition(proposal)
+  }
+
+  public async acceptProposal(
+    agentContext: AgentContext,
+    { attachmentId, proposalAttachment }: ProofFormatAcceptProposalOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const format = new ProofFormatSpec({
+      format: PRESENTATION_EXCHANGE_PRESENTATION_REQUEST,
+      attachmentId,
+    })
+
+    const presentationDefinition = proposalAttachment.getDataAsJson<PresentationDefinition>()
+
+    ps.validatePresentationDefinition(presentationDefinition)
+
+    const attachment = this.getFormatData({ presentation_definition: presentationDefinition }, format.attachmentId)
+
+    return { format, attachment }
+  }
+
+  public async createRequest(
+    agentContext: AgentContext,
+    { attachmentId, proofFormats }: FormatCreateRequestOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const presentationExchangeFormat = proofFormats.presentationExchange
+
+    if (!presentationExchangeFormat) {
+      throw Error('Missing presentation exchange format in create request attachment format')
+    }
+
+    const { presentationDefinition } = presentationExchangeFormat
+
+    ps.validatePresentationDefinition(presentationDefinition)
+
+    const format = new ProofFormatSpec({
+      format: PRESENTATION_EXCHANGE_PRESENTATION_REQUEST,
+      attachmentId,
+    })
+
+    const options = { challenge: 'TODO' }
+
+    const attachment = this.getFormatData(
+      { options, presentation_definition: presentationDefinition },
+      format.attachmentId
+    )
+
+    return { attachment, format }
+  }
+
+  public async processRequest(agentContext: AgentContext, { attachment }: ProofFormatProcessOptions): Promise<void> {
+    const ps = this.presentationExchangeService(agentContext)
+    const { presentation_definition: presentationDefinition } = attachment.getDataAsJson<{
+      presentation_definition: PresentationDefinition
+    }>()
+    ps.validatePresentationDefinition(presentationDefinition)
+  }
+
+  public async acceptRequest(
+    agentContext: AgentContext,
+    { attachmentId, requestAttachment }: ProofFormatAcceptRequestOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
```

Comment:
acceptRequest should allow user to pass which credentials they want to use for the presentation.

Reply:
We have a submission type, that would be a good input to the acceptRequest method (optional, otherwise we auto-select)

So:
- getCredentialForRequest -> returns advanced objects with all the credentials you can use for different input descriptors
- selectCredentialsForRequest -> returend submission like object that maps input descriptors to which credential(s) to use to satisfy it
- acceptRequest -> optoinally take the same structure as returned by selectCredentialsForRequest, but user can construct it themselves based on return type from getCredentialsForRequest. If not passed, call select method within the acceptRequest method